src/components/ProgressIndicators/StepBar/StepBar.tsx

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+
+import classNames from "classnames/bind";
+
+import { ArrowLink } from "../../Link";
+import styles from "./StepBar.module.scss";
+
+type StepBarProps = {
+  step: number;
+  steps: string[];
+  stepFormatter?: string;
+  onNavigate?: (i: number) => void;
+  className?: string;
+};
+
+export const StepBar: React.FC<StepBarProps> = ({ step, steps, stepFormatter, onNavigate, className }) => {
+  const translate = step >= steps.length ? `${(steps.length - 1) * 100}%` : `${(step - 1) * 100}%`;
+  const width = `${(1 / steps.length) * 100}%`;
+  const left = (i: number) => `${(i / steps.length) * 100}%`;
+
+  const goToStep = (i: number) => () => {
+    if (step - 1 >= i) onNavigate?.(i);
+  };
+
+  const text = (stepName: string, i: number) => {
+    if (stepFormatter) {
+      return stepFormatter
+        .replace(/TOTAL_STEP_COUNT/g, String(steps.length))
+        .replace(/CURRENT_STEP_INDEX/g, String(i))
+        .replace(/CURRENT_STEP_NAME/g, stepName);
+    }
```

Comment:
Does `StepBar` text formatting ever change? If it does it's probably a problem in the design system - we shouldn't have to pass this in (at least, I can't think of anywhere we should).

Reply:
Please check buy domains design and old text presentation. It is different